src/qoi/src/integrated_function.C

```diff
@@ -0,0 +1,140 @@
+//-----------------------------------------------------------------------bl-
+//--------------------------------------------------------------------------
+//
+// GRINS - General Reacting Incompressible Navier-Stokes
+//
+// Copyright (C) 2014-2016 Paul T. Bauman, Roy H. Stogner
+// Copyright (C) 2010-2013 The PECOS Development Team
+//
+// This library is free software; you can redistribute it and/or
+// modify it under the terms of the Version 2.1 GNU Lesser General
+// Public License as published by the Free Software Foundation.
+//
+// This library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+// Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc. 51 Franklin Street, Fifth Floor,
+// Boston, MA  02110-1301  USA
+//
+//-----------------------------------------------------------------------el-
+
+
+// This class
+#include "grins/integrated_function.h"
+
+// GRINS
+#include "grins/multiphysics_sys.h"
+#include "grins/assembly_context.h"
+#include "grins/materials_parsing.h"
+#include "grins/rayfire_mesh.h"
+
+// libMesh
+#include "libmesh/getpot.h"
+#include "libmesh/fem_system.h"
+#include "libmesh/quadrature_gauss.h"
+#include "libmesh/elem.h"
+#include "libmesh/fe.h"
+#include "libmesh/fe_type.h"
+#include "libmesh/function_base.h"
+#include "libmesh/fem_function_base.h"
+
+namespace GRINS
+{
+  template<typename func>
+  IntegratedFunction<func>::IntegratedFunction(unsigned int p_level, SharedPtr<func> f, SharedPtr<RayfireMesh> rayfire, const std::string& qoi_name) :
+    QoIBase(qoi_name),
+    _p_level(p_level),
+    _f(f),
+    _rayfire(rayfire)
+  {
+    // use Gauss Quadrature
+    _qbase = new libMesh::QGauss(1,libMesh::Order(_p_level));
```

Comment:
I'd do `_qbase.reset(new libMesh::QGauss(1,libMesh::Order(_p_level));` to make it a bit more clear this is a smart pointer.